packages/passport/sdk/src/Passport.test.ts

```diff
@@ -371,23 +377,45 @@ describe('Passport', () => {
   });
 
   describe('linkExternalWallet', () => {
+    const linkWalletParams = {
+      type: 'MetaMask',
+      walletAddress: '0xd8da6bf26964af9d7eed9e03e53415d37aa96045',
+      signature: 'signature123',
+      nonce: 'nonce123',
+    };
+
     it('should link external wallet when user is logged in', async () => {
-      getUserMock.mockReturnValue(mockUser);
+      getUserMock.mockReturnValue(mockUserZkEvm);
       linkExternalWalletMock.mockReturnValue(mockLinkedWallet);
 
-      const result = await passport.linkExternalWallet({
-        type: 'type', walletAddress: 'address', signature: 'signature', nonce: 'nonce',
-      });
+      const result = await passport.linkExternalWallet(linkWalletParams);
 
       expect(result).toEqual(mockLinkedWallet.data);
     });
 
     it('should throw error if user is not logged in', async () => {
       getUserMock.mockReturnValue(null);
 
-      await expect(passport.linkExternalWallet({
-        type: 'type', walletAddress: 'address', signature: 'signature', nonce: 'nonce',
-      })).rejects.toThrowError();
+      await expect(passport.linkExternalWallet(linkWalletParams)).rejects.toThrow(
+        new PassportError('User is not logged in', PassportErrorType.NOT_LOGGED_IN_ERROR),
+      );
+    });
+
+    it('should throw error if user is not registered with IMX or ZkEvm', async () => {
+      getUserMock.mockReturnValue(mockUser);
+
+      await expect(passport.linkExternalWallet(linkWalletParams)).rejects.toThrow(
+        new PassportError('User has not been registered', PassportErrorType.USER_NOT_REGISTERED_ERROR),
+      );
+    });
+
+    it('should handle errors from the linkWalletV2 API call', async () => {
```

Comment:
Can we add a couple tests to check that the status code and body of the response are still accessible here? E.g. the end user will need to be able to handle a few different error codes for 400s: (ALREADY_LINKED, MAX_WALLETS_LINKED, DUPLICATE_NONCE, VALIDATION_ERROR).

<img width="1303" alt="image" src="https://github.com/user-attachments/assets/a6393c14-7df2-487f-9573-1b62db07cd66">